src/plugins/intel_gpu/src/plugin/async_infer_request.cpp

```diff
@@ -10,45 +10,84 @@ namespace ov {
+                            std::static_pointer_cast<InferRequest>(_inferRequest)->setup_stream_graph();
+                            std::static_pointer_cast<InferRequest>(_inferRequest)->preprocess();
+                            std::static_pointer_cast<InferRequest>(_inferRequest)->enqueue();
+                            std::static_pointer_cast<InferRequest>(_inferRequest)->wait();
+            } });
+        } else {
+            _pipeline.push_back({ _waitExecutor,
+                            [this] {
+                                OV_ITT_SCOPED_TASK(itt::domains::intel_gpu_plugin, "AsyncInferRequest::WaitPipeline");
+                                std::static_pointer_cast<InferRequest>(_inferRequest)->wait_notify();
+                            } });
+        }
     }
 }
 
 void AsyncInferRequest::Infer_ThreadUnsafe() {
-    if (_inferRequest->use_external_queue()) {
-        _inferRequest->setup_stream_graph();
-        _inferRequest->preprocess_notify();
-        _inferRequest->enqueue_notify();
+    if (_isLegacy) {
+        if (std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->use_external_queue()) {
+            std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->setup_stream_graph();
+            std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->preprocess_notify();
+            std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->enqueue_notify();
+        }
+    } else {
+        if (std::static_pointer_cast<InferRequest>(_inferRequest)->use_external_queue()) {
+            std::static_pointer_cast<InferRequest>(_inferRequest)->setup_stream_graph();
+            std::static_pointer_cast<InferRequest>(_inferRequest)->preprocess_notify();
+            std::static_pointer_cast<InferRequest>(_inferRequest)->enqueue_notify();
+        }
     }
     Parent::Infer_ThreadUnsafe();
 }
 
 void AsyncInferRequest::StartAsync_ThreadUnsafe() {
-    if (_inferRequest->use_external_queue()) {
-        _inferRequest->setup_stream_graph();
-        _inferRequest->preprocess_notify();
-        _inferRequest->enqueue_notify();
+    if (_isLegacy) {
+        if (std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->use_external_queue()) {
+            std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->setup_stream_graph();
+            std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->preprocess_notify();
+            std::static_pointer_cast<InferRequestLegacy>(_inferRequest)->enqueue_notify();
+        }
+    } else {
+        if (std::static_pointer_cast<InferRequest>(_inferRequest)->use_external_queue()) {
+            std::static_pointer_cast<InferRequest>(_inferRequest)->setup_stream_graph();
+            std::static_pointer_cast<InferRequest>(_inferRequest)->preprocess_notify();
+            std::static_pointer_cast<InferRequest>(_inferRequest)->enqueue_notify();
+        }
```

Comment:
can we devide this AsyncInferRequest with Legacy and API2.0 like InferRequest?

Reply:
I reverted this commit and divide AsyncInferRequest into original and legacy as your comment. (https://github.com/andrew-k-park/openvino/commit/b905d894065bd6ab349d05a3057a8874060f1d08)